lib/radar_client.js

```diff
@@ -130,210 +132,181 @@ Client.prototype.stream = function(scope) {
   return new Scope('stream', scope, this);
 };
 
-// Access the "control" chainable operations
 Client.prototype.control = function(scope) {
   return new Scope('control', scope, this);
 };
 
+// Operations
+
 Client.prototype.nameSync = function(scope, options, callback) {
-  var message = { op: 'nameSync', to: scope };
-  if (typeof options == 'function') {
-    callback = options;
-  } else {
-    message.options = options;
-  }
-  return this._write(message, callback);
+  var request = Request.buildNameSync(scope, options);
+  return this._write(request, callback);
 };
 
 Client.prototype.push = function(scope, resource, action, value, callback) {
-  return this._write({
-    op: 'push',
-    to: scope,
-    resource: resource,
-    action: action,
-    value: value
-  }, callback);
+  var request = Request.buildPush(scope, resource, action, value);
+  return this._write(request, callback);
 };
 
 Client.prototype.set = function(scope, value, clientData, callback) {
-  var message = {
-    op: 'set',
-    to: scope,
-    value: value,
-    key: this._configuration.userId,
-    type: this._configuration.userType
-  };
-
+  var request = Request.buildSet(scope, value,
+                  this._configuration.userId, this._configuration.userType);
   if (typeof(clientData) === 'function') {
     callback = clientData;
-  } else {
-    message.clientData = clientData;
+  } else if (request) {
+    request.setAttr('clientData', clientData);
   }
 
-  return this._write(message, callback);
+  return this._write(request, callback);
 };
 
 Client.prototype.publish = function(scope, value, callback) {
-  return this._write({
-    op: 'publish',
-    to: scope,
-    value: value
-  }, callback);
+  var request = Request.buildPublish(scope, value);
+  return this._write(request, callback);
 };
 
 Client.prototype.subscribe = function(scope, options, callback) {
-  var message = { op: 'subscribe', to: scope };
-  if (typeof options == 'function') {
-    callback = options;
-  } else {
-    message.options = options;
-  }
-  return this._write(message, callback);
+  var request = Request.buildSubscribe(scope);
+  callback = _callbackOptionsSet(request, options, callback);
```

Comment:
The point here is, request.js and response.js become a common package. These options concerns should not be repeated in the api client.